src/Reddit/Reddit.ts

```diff
@@ -1,427 +1,644 @@
+        select: '.Comment',
+        expandBtn: '.icon-expand',
+      }
+    },
+    vote: {
+      btn: '.voteButton',
+      up: '.voteButton[aria-label="upvote"]',
+      down: '.voteButton[aria-label="downvote"]',
+      pressed: '.voteButton[aria-pressed="true"]',
+      unpressed: '.voteButton[aria-pressed="false"]',
+    }
+  }
+}
+
+function thingAtIndex(i: number): string {
+  if (isOldReddit) {
+    return `${ reddit.old.post.thing }[${ thingAttr }="${ i }"]`
+  } else {
+    return `${ reddit.last.post.thing }[${ thingAttr }="${ i }"]`
+  }
 }
 
 function clickIfExists(selector: string) {
-  const el = document.querySelector<HTMLElement>(selector);
-  if (el) el.click();
+  const el = select<HTMLElement>(selector)
+  if (el) el.click()
+}
+
+function clickIfDisplayed(el: HTMLElement) {
+  if (parseFloat(getComputedStyle(el).width)) {
+    el.click()
+  }
+}
+
+function genPostNumberElement(number): HTMLElement {
+  const span = document.createElement('span')
+  span.textContent = number
+  span.className = 'post-number'
+
+  setStyles({
+    position: 'absolute',
+    bottom: '2px',
+    right: '2px',
+    fontWeight: 700,
+    opacity: .3
+  }, span)
+
+  return span
+}
+
+function addOldRedditPostsAttributes(posts) {
+  posts.forEach((el) => {
+    index += 1
+    el.setAttribute(thingAttr, `${ index }`)
+    const rank = select<HTMLElement>('.rank', el)
+
+    setStyles({
+      display: 'block',
+      marginRight: '10px',
+      opacity: '1 !important'
+    }, rank!)
+  })
+}
+
+function setAttributes(post: HTMLElement) {
+  if (
+    post
+    && getComputedStyle(post).display !== 'none'
+    && !post!.querySelector('.post-number')
+  ) {
+    index += 1
+
+    post.setAttribute(thingAttr, `${ index }`)
+    post.style.position = 'relative'
+    post.appendChild(genPostNumberElement(index))
+  }
+}
+
+function addNewRedditPostsAttributes(posts) {
+  posts.forEach(setAttributes)
+}
+
+function observerCallback(mutationList) {
+  const { old, last } = reddit
+  const postSelector = isOldReddit ? old.post.thing : last.post.thing
+
+  mutationList.forEach(it => {
+    it.addedNodes.forEach(node => {
+      const post = select<HTMLElement>(postSelector, node)
+      setAttributes(post!)
+    })
+  })
+}
+
+function createObserver(el: ParentNode) {
+  observer = new MutationObserver(observerCallback)
+  observer.observe(el!, { childList: true })
+}
+
+function setParentContainer(posts) {
+  return posts![0].parentNode!.parentNode!.parentNode
+}
+
+function onLoad() {
+  if (isDOMLoaded) return
+
+  const { old, last } = reddit
+  const postSelector = isOldReddit ? old.post.thing : last.post.thing
+
+  posts = selectAll<HTMLElement>(postSelector)
+  isDOMLoaded = true
+
+  if (isOldReddit) {
+    addOldRedditPostsAttributes(posts)
+  } else {
+    scrollContainer = setParentContainer(posts)
+    createObserver(scrollContainer!)
+    addNewRedditPostsAttributes(posts)
+  }
+}
+
+function composeVoteSelector(index, cmd) {
```

Comment:
`getComposeVoteSelector` better than "compose" and same goes for the other functions that start with compose...

Also, parameters should have types.

Reply:
@wiseadme sorry about that, I meant `getVoteSelector`... compose is unnecessary in all those that you changed. Can you refactor these again?